src/CalcManager/UnitConverter.cpp

```diff
@@ -879,16 +879,26 @@ void UnitConverter::Calculate()
         else
         {
             returnValue = stod(m_returnDisplay);
-            wstring returnString;
-            if (m_currentDisplay.size() <= OPTIMALDIGITSALLOWED && abs(returnValue) >= OPTIMALDECIMALALLOWED)
+
+            auto currentDisplayTrimmed = m_currentDisplay;
+            TrimString(currentDisplayTrimmed);
+            int currentNumberSignificantDigits = static_cast<int>(currentDisplayTrimmed.size());
+            if (currentDisplayTrimmed.find(L'.') != currentDisplayTrimmed.npos)
+                --currentNumberSignificantDigits;
+            if (currentValue < 0)
+                --currentNumberSignificantDigits;
+
+            int precision = 0;
+            if (abs(returnValue) < OPTIMALDECIMALALLOWED)
             {
-                returnString = RoundSignificant(returnValue, OPTIMALDIGITSALLOWED - min(numPreDecimal, OPTIMALDIGITSALLOWED));
+                precision = MAXIMUMDIGITSALLOWED;
             }
             else
             {
-                returnString = RoundSignificant(returnValue, MAXIMUMDIGITSALLOWED - min(numPreDecimal, MAXIMUMDIGITSALLOWED));
+                precision = max(0, max(OPTIMALDIGITSALLOWED, min(MAXIMUMDIGITSALLOWED, currentNumberSignificantDigits)) - numPreDecimal);
             }
-            m_returnDisplay = returnString;
+       
+            m_returnDisplay = RoundSignificant(returnValue, precision);
```

Comment:
Is this the expected behavior in currency mode? I would think we would at least always want a precision of 2 in that mode? Left is after this change, right is before. 

![image](https://user-images.githubusercontent.com/6334170/57801332-5457e180-7708-11e9-86f4-e0ca05e3fff0.png)

Reply:
You are right, for currency, I will force the display of X decimals based on the system locale settings (while waiting for the Calculator to include the number of digit to use per currency, see https://github.com/microsoft/calculator/issues/424#issuecomment-479823338)